api_useradm.go

```diff
@@ -53,14 +67,39 @@ func (i *UserAdmHandlers) GetApp() (rest.App, error) {
 	return app, nil
 }
 
-func (i *UserAdmHandlers) AuthLoginHandler(w rest.ResponseWriter, r *rest.Request) {
-	rest.NotFound(w, r)
+func (u *UserAdmApiHandlers) AuthLoginHandler(w rest.ResponseWriter, r *rest.Request) {
+	l := requestlog.GetRequestLogger(r.Env)
+
+	//parse auth header
+	//TODO: we'll allow no auth in the 'first login' case, for now - just return 401
+	email, pass, ok := r.BasicAuth()
+	if !ok {
+		rest_utils.RestErrWithLog(w, r, l, ErrAuthHeader, http.StatusUnauthorized)
+		return
+	}
+
+	useradm, err := u.createUserAdm(config.Config, l)
```

Comment:
I'm wondering if it's possible to get rid of this pattern. The config does not change between requests. Looks like logger is the only thing left. Perhaps we could pack this into a `context.Context`? Then some helper:

```
func LoggerFromContext(ctx context.Context) {
    l, ok := ctx.Value(userKey).(*log.Logger)
    if !ok {
        return log.New(log.Ctx{})
    } 
    return l
}
```
Then when calling NewUserAdmApiHandlers() pass a closure that captures config. So the only thing left here would be (error goes away, as config is not used here anymore?):

```
useradm := u.UserAdm()
...
```